yaksh/models.py

```diff
@@ -2086,6 +2098,15 @@ def is_attempt_inprogress(self):
     def get_previous_answers(self, question):
         return self.answers.filter(question=question).order_by('-id')
 
+    def get_categorized_question_indices(self):
+
+        category_question_map = defaultdict(lambda: [])
+        for index, question in enumerate(self.get_all_ordered_questions(), 1):
+            category_question_map[
+                legend_display_types[question.type]["label"]
```

Comment:
I don't see the use of defaultdict because ```lengend_display_types``` is a normal dict and used inside a default dict so will fail in case mormal dict has ```KeyError```.

Reply:
Yes agreed.